backend/app/routers/metadata_files.py

```diff
@@ -241,6 +241,20 @@ async def update_file_metadata(
     Returns:
         Metadata document that was updated
     """
+
+    # check if metadata with file version exists, replace metadata if none exists
```

Comment:
replace_metadata is the function which is already taking care of updating resource object etc. 

If there is no specific version of metadata is available in mongodb, we want to update the version, hence called 'replace_metadata'